cabal-install/Distribution/Client/CmdExec.hs

```diff
@@ -0,0 +1,132 @@
+-------------------------------------------------------------------------------
+-- |
+-- Module      :  Distribution.Client.Exec
+-- Maintainer  :  cabal-devel@haskell.org
+-- Portability :  portable
+--
+-- Implementation of the 'new-exec' command for running an arbitrary executable
+-- in an environment suited to the part of the store built for a project.
+-------------------------------------------------------------------------------
+
+{-# LANGUAGE StandaloneDeriving #-}
+module Distribution.Client.CmdExec
+  ( execAction
+  , execCommand
+  ) where
+
+import Distribution.Client.DistDirLayout
+  ( distTempDirectory
+  )
+import Distribution.Client.Setup
+  ( GlobalFlags(..)
+  , ExecFlags(..)
+  , defaultExecFlags
+  )
+import Distribution.Client.ProjectOrchestration
+  ( ProjectBuildContext(..)
+  , PreBuildHooks(..)
+  , runProjectPreBuildPhase
+  )
+import Distribution.Client.ProjectPlanOutput
+  ( updatePostBuildProjectStatus
+  , createPackageEnvironment
+  )
+import Distribution.Simple.Command
+  ( CommandUI(..)
+  )
+import Distribution.Simple.Program.Run
+  ( ProgramInvocation(..)
+  , runProgramInvocation
+  , simpleProgramInvocation
+  )
+import Distribution.Simple.Setup
+  ( fromFlag
+  ,  optionDistPref,  optionVerbosity
+  ,  configDistPref,  configVerbosity
+  , haddockDistPref, haddockVerbosity
+  )
+import Distribution.Simple.Utils
+  ( die
+  , withTempDirectory
+  , wrapText
+  )
+
+execCommand :: CommandUI ExecFlags
+execCommand = CommandUI
+  { commandName = "new-exec"
+  , commandSynopsis = "Give a command access to the store."
+  , commandUsage = \pname ->
+    "Usage: " ++ pname ++ " new-exec [FLAGS] [--] COMMAND [--] [ARGS]\n"
+  , commandDescription = Just $ \_pname -> wrapText $
+    "TODO"
+  , commandNotes = Nothing
+  , commandDefaultFlags = defaultExecFlags
+  , commandOptions = \showOrParseArgs ->
+    [ optionVerbosity execVerbosity (\v flags -> flags { execVerbosity = v })
+    , optionDistPref
+        execDistPref (\v flags -> flags { execDistPref = v })
+        showOrParseArgs
+    ]
+  }
+
+execAction :: ExecFlags -> [String] -> GlobalFlags -> IO ()
+execAction execFlags extraArgs globalFlags = do
```

Comment:
So, one of the conventions that the other new-* commands has taken, is that they unconditionally accept ` (ConfigFlags, ConfigExFlags, InstallFlags, HaddockFlags)`. This is even the case for `new-build`, which is a change in behavior from `build` (which didn't accept certain flags.) It seems more uniform to me to follow this convention; is there a particular reason you don't here? You have a TODO below about the settings but it doesn't explain to me why you didn't just alos grab those flags.